pkg/proxy/iptables/proxier.go

```diff
@@ -581,16 +584,36 @@ func (proxier *Proxier) OnEndpointSlicesSynced() {
 // OnNodeAdd is called whenever creation of new node object
 // is observed.
 func (proxier *Proxier) OnNodeAdd(node *v1.Node) {
+	// This can not happen, the informer is filtering the node by name
 	if node.Name != proxier.hostname {
 		klog.ErrorS(nil, "Received a watch event for a node that doesn't match the current node",
 			"eventNode", node.Name, "currentNode", proxier.hostname)
+		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+	}
+	// Record the node UID if the field has not be initialized
+	// or reboot if it is different to the stored one, same Node name but different UID.
+	// https://issues.k8s.io/111321
+	proxier.mu.Lock()
+	nodeUID := string(node.UID)
+	if proxier.nodeUID == "" {
+		proxier.nodeUID = nodeUID
```

Comment:
we should log info here that we are setting to the UID. 

klog.InfoS("Setting current Node UID", "nodeUID", nodeUID)